### DISCUSSION TITLE
Permission to create table from supabase function

### DISCUSSION THREAD

**Discussion by rajatdhoot123** [supabase/supabase#14075] [Questions]
`await supabase.rpc(
          "create_table_documents_for_langchain",
          { table_name: "demo" }
        );
`

Here is my function

`BEGIN
EXECUTE format('
      CREATE TABLE IF NOT EXISTS %I (
       id serial PRIMARY KEY,
       created_at date default current_timestamp
      )', table_name);

EXECUTE format('
      ALTER TABLE %I ENABLE ROW LEVEL SECURITY;
', table_name); 
END`

Error:
`
message: 'permission denied for schema public'
`

using supabse admin key for client

**Comment by GaryAustin1** [comment]
If you mean service_role key all that does is bypass RLS.   You probably need to declare your function "security definer" type. 
This does not seem like a great idea.  Anyone can call your rpc function and create tables on your database.  You will need to provide some sort of security in the function.
